Framework/CurveFitting/src/Algorithms/NormaliseByPeakArea.cpp

```diff
@@ -283,29 +275,31 @@ double NormaliseByPeakArea::fitToMassPeak(const MatrixWorkspace_sptr &yspace,
 }
 
 /**
- * Divides the input Y & E data by the given factor
- * @param area Value to use as normalisation factor
- * @param index Index on input spectrum to normalise
- */
+* Divides the input Y & E data by the given factor
+* @param area Value to use as normalisation factor
+* @param index Index on input spectrum to normalise
+*/
 void NormaliseByPeakArea::normaliseTOFData(const double area,
                                            const size_t index) {
-  const auto &inY = m_inputWS->readY(index);
-  auto &outY = m_normalisedWS->dataY(index);
+  const auto &inY = m_inputWS->y(index);
+  std::vector<double> outY(m_normalisedWS->y(index).size());
   std::transform(inY.begin(), inY.end(), outY.begin(),
                  std::bind2nd(std::divides<double>(), area));
+  m_normalisedWS->mutableY(index) = outY;
 
-  const auto &inE = m_inputWS->readE(index);
-  auto &outE = m_normalisedWS->dataE(index);
+  const auto &inE = m_inputWS->e(index);
+  std::vector<double> outE(m_normalisedWS->e(index).size());
   std::transform(inE.begin(), inE.end(), outE.begin(),
                  std::bind2nd(std::divides<double>(), area));
+  m_normalisedWS->mutableE(index) = outE;
```

Comment:
Same here. But actually you could just use `Histogram &operator/=(Histogram &histogram, const double factor)` (include `MantidHistogramData/HistogramMath.h`) and get rid of the entire function, I think?